OpenSSL fails to build in vcpkg on MacOS when gRPC exporter is included

Failure is on MacOS only and appears to be caused by a recent update in GitHub runners to use CMake 4.0.

The error log looks like this:
```
clang: warning: no such sysroot directory: '-g' [-Wmissing-sysroot]
clang: warning: no such sysroot directory: '-g' [-Wmissing-sysroot]
clang: warning: no such sysroot directory: '-g' [-Wmissing-sysroot]
clang: warning: no such sysroot directory: '-g' [-Wmissing-sysroot]
../src/nssl-3.3.1-6a4778eae7.clean/crypto/aes/aes_core.c:46:10: fatal error: 'assert.h' file not found
#include <assert.h>
         ^~~~~~~~~~
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/crypto/aes/aes_cfb.c:17:
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/include/openssl/modes.h:20:
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/include/openssl/types.h:32:
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/include/openssl/e_os2.h:235:
/Applications/Xcode_15.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/inttypes.h:21:15: fatal error: 'inttypes.h' file not found
#include_next <inttypes.h>
              ^~~~~~~~~~~~
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/crypto/aes/aes_cbc.c:18:
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/include/openssl/modes.h:20:
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/include/openssl/types.h:32:
In file included from ../src/nssl-3.3.1-6a4778eae7.clean/include/openssl/e_os2.h:235:
/Applications/Xcode_15.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/15.0.0/include/inttypes.h:21:15: fatal error: 'inttypes.h' file not found
#include_next <inttypes.h>
              ^~~~~~~~~~~~
1 error generated.
1 error generated.
make[1]: *** [crypto/aes/libcrypto-lib-aes_cbc.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [crypto/aes/libcrypto-lib-aes_cfb.o] Error 1
../src/nssl-3.3.1-6a4778eae7.clean/crypto/aes/aes_ecb.c:10:10: fatal error: 'assert.h' file not found
#include <assert.h>
         ^~~~~~~~~~
1 error generated.
make[1]: *** [crypto/aes/libcrypto-lib-aes_ecb.o] Error 1
1 error generated.
make[1]: *** [crypto/aes/libcrypto-lib-aes_core.o] Error 1
make: *** [build_sw] Error 2
```
From this discussion, a user suggests upgrading to vcpkg 2025.04.09 or higher
[https://github.com/microsoft/vcpkg/issues/44229](https://github.com/microsoft/vcpkg/issues/44229)

--- a/.github/workflows/build_and_test_full.yml
+++ b/.github/workflows/build_and_test_full.yml
@@ -148,6 +148,8 @@ jobs:
                 products: MATLAB_Compiler MATLAB_Compiler_SDK
             - name: Build OpenTelemetry-Matlab
               working-directory: opentelemetry-matlab
+              env:
+                 CMAKE_POLICY_VERSION_MINIMUM: 3.5   # required by upb because its cmake requirement is not compatible with cmake 4 
               run: |
                   cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DWITH_EXAMPLES=ON -DWITH_OTLP_GRPC=ON -DUSE_BATCH_FOR_MCC=ON -DOTEL_MATLAB_VERSION=${{ needs.get_version.outputs.version }} -DCMAKE_INSTALL_PREFIX=${{ env.OPENTELEMETRY_MATLAB_INSTALL }}
                   cmake --build build --config Release --target install

--- a/.github/workflows/create_release.yml
+++ b/.github/workflows/create_release.yml
@@ -85,6 +85,8 @@ jobs:
                 products: MATLAB_Compiler
             - name: Build OpenTelemetry-Matlab
               working-directory: opentelemetry-matlab
+              env:
+                 CMAKE_POLICY_VERSION_MINIMUM: 3.5   # required by upb because its cmake requirement is not compatible with cmake 4 
               run: |
                   cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DWITH_OTLP_GRPC=ON -DWITH_OTLP_FILE=ON -DOTEL_MATLAB_VERSION=${{ github.ref_name }} -DCMAKE_INSTALL_PREFIX=${{ env.OPENTELEMETRY_MATLAB_INSTALL }}
                   cmake --build build --config Release --target install

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ else()
 
    set(VCPKG_FETCH_CONTENT_NAME vcpkg)
    set(VCPKG_GIT_REPOSITORY "https://github.com/microsoft/vcpkg.git")
-   set(VCPKG_GIT_TAG "b02e341")
+   set(VCPKG_GIT_TAG "ce613c4")
    FetchContent_Declare(
        ${VCPKG_FETCH_CONTENT_NAME}
        GIT_REPOSITORY ${VCPKG_GIT_REPOSITORY}

--- a/vcpkg.json
+++ b/vcpkg.json
@@ -20,7 +20,10 @@
         "abseil",
         "c-ares",
         "re2",
-        "openssl",
+	{
+	  "name": "openssl",
+	  "version>=": "3.5.0"
+	},
         "upb"
       ]
     }

